### ISSUE TITLE
Using memcached client with large values truncates response

### ISSUE THREAD

**Issue by samfrench** [comandeo/crystal-memcached#17]
Hi @comandeo, thank you for fixing https://github.com/comandeo/crystal-memcached/issues/9 previously. I have been looking again at getting and setting large values in memcached and it does look like there is still an issue. All the specs are fine and pass for me, but when I use this shard in my project, I always get a truncated response.

E.g.

File called example.cr
```cr
require "memcached"

client = Memcached::Client.new("127.0.0.1", 11211)
value = "LargeValue" * 1000
client.set("LargeKey", value)
response = client.get("LargeKey")
puts response.to_s.bytesize
puts response
```

Then I run:
```
crystal run example.cr
```

The output I get:
```
10000
LargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLargeValueLarg
```

I would expect 10000 characters to be output as "LargeValue" is 10 characters and multiplied by 1000 should give me 10000. For some reason I only get 8164 characters. Before setting the item in the cache I have verified that is the full 10000 characters. When looking into it, it doesn't always seem to be 8164 characters but I don't get the full response. When checking the bytesize of the string, it suggests that the string is 10000 characters but I don't see them.

I tried to add a spec for this, but it seems to only be an issue when running it as part of an application. I did wonder if it was the memcached image I was using, but as the specs pass with it, I think it is only when retrieving the item from the cache where it gets truncated. Any help in this area would be appreciated.

**Comment by samfrench** [comment]
Hi @comandeo, can you replicate this issue?

**Comment by comandeo** [comment]
Hi @samfrench ,

I am very sorry, but I didn't have a chance to check this issue out. I do not use crystal in any of my projects anymore. Therefore, I can work on this client only int my free time, and I do not have a lot at the moment. I hope I will be able to work on this issue later this week.

Sorry again for the delay.

**Comment by samfrench** [comment]
Hi @comandeo, that's fine. I was just checking to see if this hadn't been noticed. As you saw it then I assumed this was a side project and wasn't something worked on day to day. Thank you for the reply.

**Comment by comandeo** [comment]
Hi @samfrench ,

It seems that I cannot reproduce the issue. I just created a new project, added "crystal-memcached" dependency, and ran your example. I got exactly 10000 characters output. 

So, now it is even more weird. Could you please check shards.lock file of your project? It should refer to version 0.2.1 of the client.

**Comment by samfrench** [comment]
Hi @comandeo. Looking at the shards.lock file I do think this was the cause and I cannot replicate the issue when this is fixed to 0.2.1. Thank you for assisting with this. This can be closed.
